Modules/Core/Common/include/itkMacro.h

```diff
@@ -77,77 +77,60 @@ namespace itk
+  #else
+    #define CLANG_SUPPRESS_Wc__14_extensions
+  #endif
 #else
-#define CLANG_PRAGMA_PUSH
-#define CLANG_PRAGMA_POP
-#define CLANG_SUPPRESS_Wfloat_equal
-#define CLANG_SUPPRESS_Wc__14_extensions
+  #define CLANG_PRAGMA_PUSH
+  #define CLANG_PRAGMA_POP
+  #define CLANG_SUPPRESS_Wfloat_equal
+  #define CLANG_SUPPRESS_Wc__14_extensions
 #endif
 
 // Intel compiler convenience macros
 #if defined(__INTEL_COMPILER)
-#define INTEL_PRAGMA_WARN_PUSH ITK_PRAGMA(warning push)
-#define INTEL_PRAGMA_WARN_POP  ITK_PRAGMA(warning pop)
-#define INTEL_SUPPRESS_warning_1292  ITK_PRAGMA(warning disable 1292)
+  #define INTEL_PRAGMA_WARN_PUSH ITK_PRAGMA(warning push)
+  #define INTEL_PRAGMA_WARN_POP  ITK_PRAGMA(warning pop)
+  #define INTEL_SUPPRESS_warning_1292  ITK_PRAGMA(warning disable 1292)
 #else
-#define INTEL_PRAGMA_WARN_PUSH
-#define INTEL_PRAGMA_WARN_POP
-#define INTEL_SUPPRESS_warning_1292
+  #define INTEL_PRAGMA_WARN_PUSH
+  #define INTEL_PRAGMA_WARN_POP
+  #define INTEL_SUPPRESS_warning_1292
 #endif
 
 // Define ITK_GCC_PRAGMA_DIAG(param1 [param2 [...]]) macro.
 //
-// This macros sets a pragma diagnostic if it supported by the version
-// of GCC being used otherwise it is a no-op.
+// This macro sets a pragma diagnostic
 //
-// GCC diagnostics pragma supported only with GCC >= 4.2
-#if defined( __GNUC__ ) && !defined( __INTEL_COMPILER )
-#  if ( __GNUC__ > 4 ) || (( __GNUC__ >= 4 ) && ( __GNUC_MINOR__ >= 2 ))
-#    define ITK_GCC_PRAGMA_DIAG(x) ITK_PRAGMA(GCC diagnostic x)
-#  else
-#    define ITK_GCC_PRAGMA_DIAG(x)
-#  endif
-#else
-#  define ITK_GCC_PRAGMA_DIAG(x)
-#endif
-
 // Define ITK_GCC_PRAGMA_DIAG_(PUSH|POP) macros.
 //
 // These macros respectively push and pop the diagnostic context
-// if it is supported by the version of GCC being used
-// otherwise it is a no-op.
 //
-// GCC push/pop diagnostics pragma are supported only with GCC >= 4.6
-//
-// Define macro ITK_HAS_GCC_PRAGMA_DIAG_PUSHPOP if it is supported.
+// Define macro ITK_HAS_GCC_PRAGMA_DIAG_PUSHPOP
 #if defined( __GNUC__ ) && !defined( __INTEL_COMPILER )
-#  if ( __GNUC__ > 4 ) || (( __GNUC__ >= 4 ) && ( __GNUC_MINOR__ >= 6 ))
-#    define ITK_GCC_PRAGMA_DIAG_PUSH() ITK_GCC_PRAGMA_DIAG(push)
-#    define ITK_GCC_PRAGMA_DIAG_POP() ITK_GCC_PRAGMA_DIAG(pop)
-#    define ITK_HAS_GCC_PRAGMA_DIAG_PUSHPOP
-#  else
-#    define ITK_GCC_PRAGMA_DIAG_PUSH()
-#    define ITK_GCC_PRAGMA_DIAG_POP()
-#  endif
+  #define ITK_GCC_PRAGMA_DIAG(x) ITK_PRAGMA(GCC diagnostic x)
+  #define ITK_GCC_PRAGMA_DIAG_PUSH() ITK_GCC_PRAGMA_DIAG(push)
+  #define ITK_GCC_PRAGMA_DIAG_POP() ITK_GCC_PRAGMA_DIAG(pop)
+  #define ITK_HAS_GCC_PRAGMA_DIAG_PUSHPOP
 #else
-#  define ITK_GCC_PRAGMA_DIAG_PUSH()
-#  define ITK_GCC_PRAGMA_DIAG_POP()
+  #define ITK_GCC_PRAGMA_DIAG(x)
+  #define ITK_GCC_PRAGMA_DIAG_PUSH()
+  #define ITK_GCC_PRAGMA_DIAG_POP()
 #endif
 
 /*
- * ITK only supports MSVC++ 7.1 and greater
+ * ITK only supports MSVC++ 10.0 and greater
```

Comment:
Since we require C++11, we need at least VS2013. But practically, that version cannot compile ITKv5 due to its limited support of C++11. So the minimum supported version is VS2015 aka `_MSC_VER` 1900. See [this](https://devblogs.microsoft.com/cppblog/side-by-side-minor-version-msvc-toolsets-in-visual-studio-2017/).